code/script.js

```diff
@@ -30,18 +55,134 @@ const showMessage = (message, sender) => {
+    showMessage("Based on your preferences;<br>You're a Couch potato who live in a city or suburb and wants a small sized dog with a long coat.<br> You should get a short coated Chihuahua or a Pekingese!", "bot")
+  } else {
+  showMessage("I need to think about this a bit, come back later!", "bot")
+  }
+}
+
+//Interaction 6, confirm environment choice and ask what coat they prefer on a dog. Store value.
+const coatChoice = (live) => {
+  showMessage(`Ok, ${live} it is! Do you want a dog with a short or long coat?`, "bot")
+    inputWrapper.innerHTML = buttonSelectCoat
+    document
+      .getElementById("long")
+      .addEventListener("click", () => {
+      showMessage("I want a dog with a long coat", "user")
+      inputWrapper.innerHTML = ""
+      setTimeout(() => dogBreed("long"), 1000)
+  })
+  document
+      .getElementById("short")
+      .addEventListener("click", () => {
+      showMessage("I want a dog with a short coat", "user")
+      inputWrapper.innerHTML = ""
+      setTimeout(() => dogBreed("short"), 1000)
+  })
+  const coat = document.getElementById("submit")
+}
+
+//Interaction 5, confirm size choice and ask if they live in city/suburb or countryside. Store value.
+const environmentChoice = (size) => {
+  showMessage(`Ok, you you want a ${size} sized dog breed. Where do you live?`, "bot")
+  inputWrapper.innerHTML = buttonSelectEnvironment
+  document
+    .getElementById("citySuburb")
+    .addEventListener("click", () => {
+    showMessage("I live in a city or suburb", "user")
+    inputWrapper.innerHTML = ""
+    setTimeout(() => coatChoice("citySuburb"), 1000)
+  })
+    document
+    .getElementById("countryside")
+    .addEventListener("click", () => {
+    showMessage("I live in the countryside", "user")
+    inputWrapper.innerHTML = ""
+    setTimeout(() => coatChoice("countryside"), 1000)
+  })
+  const live = document.getElementById("submit")
+}
+
+//Interaction 4, confirm lifestyle choice and ask for size. Store value.
+const lifestyleChoice = (specificChoice) => {
+  if(specificChoice === "couchPotato") {
+    showMessage("Ok, so you like to take it easy. You need a companion that likes to do the same. All dogs needs exercise though, but some less than others. What size of dog do you like?", "bot")
+    inputWrapper.innerHTML = selectSize
+  } else if (specificChoice === "active") {
+    showMessage("Ok, so you like to move! Let's find you a companion that likes it just as much. What size of dog do you like?", "bot")
+    inputWrapper.innerHTML = selectSize
+  } else {
+    showMessage("You should probably think twice about getting a dog. No dog likes a human with no time.", "bot")
+  }
+  const size = document.getElementById("select")
+  addEventListener("change", () => {
+    showMessage(`I like ${size.value} sized dogs.`, "user")
+    inputWrapper.innerHTML = ""
+    setTimeout(() => environmentChoice(size.value), 1000)
```

Comment:
**4.** If you were to pass along the arguments instead of using the global variables, you would have to pass along both `specificChoice` and the size here:
`environmentChoice(specificChoice, size.value)`